js/components/accessibilityControls.js

```diff
@@ -61,6 +73,41 @@ AccessibilityControls.prototype = {
     }
   },
 
+  keyEventUp: function(e) {
+    if (!this.controller.state.accessibilityControlsEnabled) { return; }
+    var targetTagName = this.getTargetTagName(e);
+    var charCode = e.which || e.keyCode;
+    this.handleKeyClick(e, this.state.keyMapsList, charCode, false, targetTagName);  //stop rotate 360
+  },
+
+  getTargetTagName: function(e) {
+    var targetTagName = "";
+    if (e.target && typeof e.target.tagName === "string") {
+      targetTagName = e.target.tagName.toLowerCase();
+    }
+    return targetTagName;
+  },
+
+  handleKeyClick: function(e, keyMapsList, char, bool, targetTagName) {
+    /*
+     * keyMapsList - array of objects {char: 83, direction: 'down'}
+     */
+    if (!this.controller.videoVr) { return; }
+    for (var i=0; i<keyMapsList.length; i++) {
```

Comment:
It looks like we can avoid this loop by using the hashmap suggestion above.

Reply:
\+